src/openfermion/utils/_davidson.py

```diff
@@ -0,0 +1,273 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""This module is to find lowest eigenvalues with Davidson algorithm."""
+from __future__ import absolute_import
+
+import numpy
+import numpy.linalg
+import scipy
+import scipy.linalg
+import scipy.sparse.linalg
+
+from openfermion.utils._sparse_tools import (get_linear_qubit_operator,
+                                             get_linear_qubit_operator_diagonal)
+
+# Exceptions.
+class DavidsonError(Exception):
+    """Exceptions."""
+    pass
+
+
+class Davidson(object):
+    """Davidson algorithm to get the n states with smallest energies."""
+
+    def __init__(self, linear_operator, linear_operator_diagonal, eps=1e-6):
```

Comment:
If I recall correctly, Davidson also has another parameter which is the number of vectors to use in the subspace solves, should we pass that in here as well?

Reply:
Done, there was a TODO comment in the previous version, now removed.